File: DS/LinkedList/PrintReverseLinkedList.cs
//Print Linked List in reserve order.

using System;

namespace ConsoleApplication1
{
    class Nodes
    {
        public int data;
        public Nodes nextnode;
        public Nodes(int _data, Nodes _nextnode)
        {
            this.data = _data;
            this.nextnode = _nextnode;
        }
    }
    class LinkedList
    {
        Nodes currnode;

        public LinkedList()
        {
            this.currnode = null;
        }

        void Add(int data)
        {
            Nodes _nodes = new Nodes(data, this.currnode);
            this.currnode = _nodes;
        }        

        void PrintReverse(Nodes head)
        {
            if(head!=null)
            {
                PrintReverse(head.nextnode);
                Console.WriteLine(head.data);
            }
        }
    }
}

File: DS/LinkedList/PrintReverseLinkedList.py
"""
Print Linked List in reserve order.
We need to get to end of the linked list by calling recurvise function with parameter next node.
And then please data of each node.
"""

class Node:
    def __init__(self,data,next):
        self.data = data
        self.next = next

class LinkedList:
    def __init__ (self):
        self.currnode = None

    def add(self,data):
        newNode = Node(data,self.currnode)
        self.currnode = newNode
        return self.currnode
    
    def printR(self,head):
        print 'Start Fun'
        if head:
            print 'Before recurvise'
            self.printR(head.next)
            print 'After recurvise'
            print head.data
       
if __name__ == "__main__":
    li = LinkedList()
    li.add(2)
    li.add(4)
    node = li.add(5)
    li.printR(node)